pandas/tools/plotting.py

```diff
@@ -1217,8 +1217,22 @@ def _adorn_subplots(self):
 
         if self.title:
             if self.subplots:
-                self.fig.suptitle(self.title)
+                if is_list_like(self.title):
+                    if len(self.title) != len(self.axes):
```

Comment:
pls use  ``self.nseries``. Number of columns and axes and can differ when ``layout`` is specified.

Also add test case with 3 numeric columns df with ``df.plot(subplots=True, layout=(2, 2))``